### ISSUE TITLE
Does this project support NVIDIA's T4 GPU ？

### ISSUE THREAD

**Issue by zhongwenkun886** [nunchaku-tech/nunchaku#543]
When I run the following code on a NVIDIA T4 GPU, it prompts: "This kernel is not supported on your GPU."

import time
import torch
from diffusers import FluxKontextPipeline
from PIL import Image

from nunchaku.models.transformers.transformer_flux import NunchakuFluxTransformer2dModel

precision = "int4"

pipeline_init_kwargs = {}
transformer = NunchakuFluxTransformer2dModel.from_pretrained(
    f"./models/svdq-{precision}_r32-flux.1-kontext-dev.safetensors", torch_dtype=torch.float16,
    offload=True
)
pipeline_init_kwargs["transformer"] = transformer

from nunchaku.models.text_encoders.t5_encoder import NunchakuT5EncoderModel

text_encoder_2 = NunchakuT5EncoderModel.from_pretrained(
    "./models/mit-han-lab/nunchaku-t5/awq-int4-flux.1-t5xxl.safetensors", torch_dtype=torch.float16
)
pipeline_init_kwargs["text_encoder_2"] = text_encoder_2

pipeline = FluxKontextPipeline.from_pretrained(
    "./models/FLUX.1-Kontext-dev", torch_dtype=torch.float16, **pipeline_init_kwargs
)

pipeline = pipeline.to("cuda")
pipeline.precision = precision

img_path = "assets/pexels-photo.jpeg"
prompt = "Change the color of the woman's dress to red. The background is a beach. The woman is holding a sign that says 'Nunchaku is awesome."
num_inference_steps = 20
guidance_scale = 2.5
seed = 233

img = Image.open(img_path).convert('RGB')

result_image = pipeline(
    prompt=prompt,
    image=img,
    height=img.height,
    width=img.width,
    num_inference_steps=num_inference_steps,
    guidance_scale=guidance_scale,
    generator=torch.Generator().manual_seed(seed),
).images[0]

**Comment by azoksky** [comment]
Unfortunately no..

**Comment by lmxyy** [comment]
It should be supported as it is Turing GPUs. What issue did you meet?

**Comment by zhongwenkun886** [comment]
> It should be supported as it is Turing GPUs. What issue did you meet?
When I run the above code, an error was prompted: "This kernel is not supported on your GPU." Upon further debugging of the code, I found that this error occurred on line 86 of the code file nunchaku\models\text_encoders\linear.py:
out = gemm_awq(x, self.qweight, self.scales, self.scaled_zeros)

**Comment by zhongwenkun886** [comment]
> It should be supported as it is Turing GPUs. What issue did you meet?

When I run the above code, an error was prompted: "This kernel is not supported on your GPU." Upon further debugging of the code, I found that this error occurred on line 86 of the code file nunchaku\models\text_encoders\linear.py:

out = gemm_awq(x, self.qweight, self.scales, self.scaled_zeros)

and all the parameters in the function "gemm_awq" are torch.float16.

**Comment by lmxyy** [comment]
Also see [https://github.com/nunchaku-tech/nunchaku/issues/420](https://github.com/nunchaku-tech/nunchaku/issues/420)

**Comment by tayton42** [comment]
I have the same issue.Did you fix this problem?
